packages/block-library/src/video/video-player.native.js

```diff
@@ -18,32 +18,40 @@ import styles from './video-player.scss';
 class Video extends Component {
 	constructor() {
 		super( ...arguments );
+		this.isIOS = Platform.OS === 'ios';
 		this.state = {
-			isLoaded: false,
 			isFullScreen: false,
 		};
 		this.onPressPlay = this.onPressPlay.bind( this );
-		this.onLoad = this.onLoad.bind( this );
-		this.onLoadStart = this.onLoadStart.bind( this );
-	}
-
-	onLoad() {
-		this.setState( { isLoaded: true } );
-	}
-
-	onLoadStart() {
-		this.setState( { isLoaded: false } );
 	}
 
 	onPressPlay() {
-		if ( this.player ) {
-			this.player.presentFullscreenPlayer();
+		if ( this.isIOS ) {
+			if ( this.player ) {
+				this.player.presentFullscreenPlayer();
+			}
+		} else {
+			const { source } = this.props;
+			if ( source && source.uri ) {
+				this.openURL( source.uri )
+			}
 		}
 	}
 
+	// Tries opening the URL outside of the app
+	openURL( url ) {
+		Linking.canOpenURL(url).then(( supported ) => {
+			if ( !supported ) {
+				console.warn("Can't open the video URL: " + url);
```

Comment:
On Android there is also an option to present the bottom sheet with the list of apps that can open the video, maybe that could be a good fall back scenario?

Reply:
Got the explanation from @pinarol, so we are good with the current solution.

Reply:
AFAIU, if there are apps that can open the url then `supported ` flag won't be false ever. https://facebook.github.io/react-native/docs/linking#canopenurl

Reply:
Btw, on WPAndroid Aztec, when Chrome is disabled, it pops up the message for the user: "No application can handle this request. Please install a Web browser", maybe we can do the same so that user knows the reason why video can't be opened?

Reply:
Sounds good 👍 added here https://github.com/WordPress/gutenberg/pull/16089/commits/941917fad722b7d83d990d4dd147f63b55e59f7f